Bottom: 118cfab2efb6837f6f2af57eed7909c888e99168
Top:    826d0da7f58b1f81aed5678c5c3d8e0f0aef6be1
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-01-10 15:54:44 +0800

Refresh of tun-allow-attach-ebpf-filter

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index c0ba627..170a3e8 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1088,11 +1088,10 @@ static netdev_tx_t tun_net_xmit(struct sk_buff *skb, struct net_device *dev)
 
 	len = run_ebpf_filter(tun, skb, len);
 
-	/* Trim extra bytes since we may inster vlan proto & TCI
+	/* Trim extra bytes since we may insert vlan proto & TCI
 	 * in tun_put_user().
 	 */
-	if (skb_vlan_tag_present(skb))
-		len -= skb_vlan_tag_present(skb) ? sizeof(struct veth) : 0;
+	len -= skb_vlan_tag_present(skb) ? sizeof(struct veth) : 0;
 	if (len <= 0 || pskb_trim(skb, len))
 		goto drop;
